src/App.tsx

```diff
@@ -1,25 +1,54 @@
 import React from 'react';
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
 
 import './App.scss';
 import Start from './components/Start';
-import { Finish } from './components/Finish';
+import { TodoList } from './components/TodoList';
+import { SortPanel } from './components/SortPanel';
 
-import { isLoading, getMessage } from './store';
+import {
+  startLoading,
+  isLoading,
+  finishLoading,
+  getMessage,
+  setTodos,
+  setVisibleContent,
+  getVisibleContent,
+} from './store';
 
+import { getTodosFromServer } from './helpers/api';
 
 const App = () => {
+  const dispatch = useDispatch();
   const loading = useSelector(isLoading);
-  const message = useSelector(getMessage) || 'Ready!';
+  const message = useSelector(getMessage) || '';
+  const visibleContent = useSelector(getVisibleContent);
+
+  const loadTodos = async () => {
+    dispatch(startLoading());
+    try {
+      const todosFromServer = await getTodosFromServer();
+
+      dispatch(setTodos(todosFromServer));
+      dispatch(finishLoading(''));
```

Comment:
what does this method `finishLoading` should do, I can't understand from its name

Reply:
finishLoading  shows that loading are finished and we can see result

Reply:
then name it as setFinishResult because it sets some message and has nothing in common with loading